### PR TITLE
Upgrade to ML.NET 0.7

### LINKED ISSUE #54: Upgrade to ML.NET 0.7
Currently NimbusML runs on ML.NET 0.6 version.
With the recent ML.NET release of 0.7 https://github.com/dotnet/machinelearning/releases/tag/v0.7.0 
 it would be great to get the latest ML.NET bits

### PR DESCRIPTION
**Pull request by ganik**
Upgrade to ML.NET 0.7, fixes #54 :
 * updated manifest.json
 * regenerated entrypoints and classes
 * updated manifest generator project

### CODE REVIEW

**Comment by ganik** [comment]
2 types of errors left:
1. Error: *** System.DllNotFoundException: 'Unable to load shared library 'lib_lightgbm' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(liblib_lightgbm, 1): image not found'

2. For all Crossvalidation tests:
       Error: *** System.ArgumentNullException: 'Value cannot be null.
       Parameter name: estimator'

First error is only Mac specific. And reminds me of past issue where lib_lightgbm was not built using correct gcc compiler. @montebhoover  could you look into this pls
I ll continue investigating the second one.

File: src/python/nimbusml/internal/core/preprocessing/schema/columnselector.py [hunk 1]
```diff
@@ -65,8 +83,15 @@ def _get_node(self, **all_args):
                 "input has to be a list of strings, instead got %s" %
                 type(input_columns))
 
+        keep_columns = self.keep_columns
+        if self.keep_columns is None and self.drop_columns is None:
```
Comment (montebhoover) [inline src/python/nimbusml/internal/core/preprocessing/schema/columnselector.py]: I made an issue discussing how to deal with the change in the ML.NET entrypoint: https://github.com/Microsoft/NimbusML/issues/76

**Comment by montebhoover** [comment]
> # - Generated by tools/entrypoint_compiler.py: do not edit by hand

Do you mind if we change this line to make a note that this is not autogenerated and is required for backward compatibility?

---
Refers to: src/python/nimbusml/internal/core/preprocessing/schema/columndropper.py:5 in d23d696. [](commit_id = d23d696d0fd4ae3eafb3eb9b9d70f596dc46adac, deletion_comment = False)

**Comment by montebhoover** [comment]
>     "AffinityCalc.Calculate",

Everything in this list is no longer an entrypoint in ML.NET, so they can be removed.  They don't hurt anything for now though, so maybe I should make a separate PR to remove them.

---
Refers to: src/python/tools/manifest_diff.json:164 in d23d696. [](commit_id = d23d696d0fd4ae3eafb3eb9b9d70f596dc46adac, deletion_comment = False)

File: src/python/nimbusml/internal/core/preprocessing/schema/columnselector.py (hunk 1 above)
Comment (montebhoover) [reply src/python/nimbusml/internal/core/preprocessing/schema/columnselector.py]: Nice fix @ 

---
In reply to: [243136165](https://github.com/Microsoft/NimbusML/pull/55#discussion_r243136165) [](ancestors = 243136165)

**Comment by montebhoover** [comment]
>     "AffinityCalc.Calculate",

Also, we can delete the file `transforms_wordtokenizer.py` in entrypoints, as that is no longer in manifest.json.

---
In reply to: [449106410](https://github.com/Microsoft/NimbusML/pull/55#issuecomment-449106410) [](ancestors = 449106410)

---
Refers to: src/python/tools/manifest_diff.json:164 in d23d696. [](commit_id = d23d696d0fd4ae3eafb3eb9b9d70f596dc46adac, deletion_comment = False)

File: src/python/nimbusml/internal/entrypoints/timeseriesprocessing_iidchangepointdetector.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# - Generated by tools/entrypoint_compiler.py: do not edit by hand
```
Comment (montebhoover) [inline src/python/nimbusml/internal/entrypoints/timeseriesprocessing_iidchangepointdetector.py]: It looks like we didn't generate a public API for these entrypoints that got added.  Maybe we should add that in a separate PR?

Comment (ganik) [reply src/python/nimbusml/internal/entrypoints/timeseriesprocessing_iidchangepointdetector.py]: Yes, adding them to be public facing should be done as a separate task. Currently by default all entrypoints are generated but only the ones specified in manifest_diff.json will have user facing interface / class

---
In reply to: [243395603](https://github.com/Microsoft/NimbusML/pull/55#discussion_r243395603) [](ancestors = 243395603)

**Comment by ganik** [comment]
>     "AffinityCalc.Calculate",

the transform_wordtokenizer is still part of manifest. 

---
In reply to: [449112798](https://github.com/Microsoft/NimbusML/pull/55#issuecomment-449112798) [](ancestors = 449112798,449106410)

---
Refers to: src/python/tools/manifest_diff.json:164 in d23d696. [](commit_id = d23d696d0fd4ae3eafb3eb9b9d70f596dc46adac, deletion_comment = False)

**Pull request merged**